FirebaseInAppMessaging/Sources/Analytics/FIRIAMClearcutLogStorage.m

```diff
@@ -168,7 +168,16 @@ - (void)loadFromCachePath:(NSString *)cacheFilePath {
   NSTimeInterval start = [self.timeFetcher currentTimestampInSeconds];
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-  id fetchedClearcutRetryRecords = [NSKeyedUnarchiver unarchiveObjectWithFile:filePath];
+  id fetchedClearcutRetryRecords;
+    NSData *data = [NSData dataWithContentsOfFile:filePath];
+        if (data) {
+            if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, *)) {
+                fetchedClearcutRetryRecords = [NSKeyedUnarchiver unarchivedObjectOfClass:[NSMutableArray<FIRIAMClearcutLogRecord *> class] fromData:data error:nil];
+            } else {
+                // Fallback on earlier versions
+                fetchedClearcutRetryRecords = [NSKeyedUnarchiver unarchiveObjectWithFile:filePath];
```

Comment:
Lines 169, 170, and 181 serve to prevent the warnings (like from deprecated API) from showing. You can now reduce the code that warnings are disabled for to only line 178 because the deprecated `+[NSKeyedUnarchiver unarchiveObjectWithFile:]` API is the only line that needs to be surrounded with the `#pragma clang` stuff.

- Remove 169, 170, and 181
- Add:
```suggestion
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
                fetchedClearcutRetryRecords = [NSKeyedUnarchiver unarchiveObjectWithFile:filePath];
#pragma clang diagnostic pop
```

Reply:
Got it! I'm done with the changes.